### PR TITLE
less colors and more contrast in terminal

### LINKED ISSUE #6512: Improve repl contrast on (very) dark terminal background
In real life the contrast is worse than in the photo. The blue text is unreadable and the pink colon's visibility is not good enough.

### PR DESCRIPTION
**Pull request by Anton-4**
I checked how our terminal colors looked on macos. ubuntu, nixOS and raspberryOS. The colors we internally call blue and magenta regularly did not meet the AA wcag standard so I removed them and used other colors instead. I also changed the colors we still use to their "improved visibility variants". Other popular languages and command line tools often use these variants as well.

after (left) and before (right) on raspberryOS:

pi_repl_screenshot

Closes #6512

### CODE REVIEW

File: crates/reporting/src/report.rs [hunk 1]
```diff
@@ -260,12 +258,10 @@ pub struct StyleCodes {
...
-    red: "\u{001b}[31m",
-    green: "\u{001b}[32m",
-    yellow: "\u{001b}[33m",
-    blue: "\u{001b}[34m",
-    magenta: "\u{001b}[35m",
-    cyan: "\u{001b}[36m",
+    red: "\u{001b}[1;31m",
+    green: "\u{001b}[1;32m",
+    yellow: "\u{001b}[1;33m",
+    cyan: "\u{001b}[1;36m",
```
Comment (Anton-4) [inline crates/reporting/src/report.rs:264]: `1;` to use the increased visibility variants.

File: crates/cli/src/lib.rs [hunk 2]
```diff
@@ -541,7 +542,7 @@ pub fn test(matches: &ArgMatches, triple: Triple) -> io::Result<i32> {
 
     let mut writer = std::io::stdout();
 
-    let (failed, passed) = roc_repl_expect::run::run_toplevel_expects(
+    let (failed_count, passed_count) = roc_repl_expect::run::run_toplevel_expects(
```
Comment (Anton-4) [inline crates/cli/src/lib.rs:545]: More self-evident names

File: crates/cli/src/main.rs [hunk 3]
```diff
@@ -210,33 +210,7 @@ fn main() -> io::Result<()> {
...
-                        problems.warnings,
-                        if problems.warnings == 1 {
-                            "warning"
-                        } else {
-                            "warnings"
-                        },
-                        total_time.as_millis(),
-                    );
-
+                    problems.print_to_stdout(total_time);
```
Comment (Anton-4) [inline crates/cli/src/main.rs:213]: This code was duplicated in `print_to_stdout`.

File: crates/cli_utils/src/helpers.rs [hunk 4]
```diff
@@ -98,22 +98,6 @@ pub fn path_to_binary(binary_name: &str) -> PathBuf {
...
-        .replace(ANSI_STYLE_CODES.blue, "")
-        .replace(ANSI_STYLE_CODES.magenta, "")
-        .replace(ANSI_STYLE_CODES.cyan, "")
-        .replace(ANSI_STYLE_CODES.white, "")
-        .replace(ANSI_STYLE_CODES.bold, "")
-        .replace(ANSI_STYLE_CODES.underline, "")
-        .replace(ANSI_STYLE_CODES.reset, "")
-        .replace(ANSI_STYLE_CODES.color_reset, "")
-}
-
```
Comment (Anton-4) [inline crates/cli_utils/src/helpers.rs:116]: This was moved.

**Comment by rtfeldman** [comment]
Very cool! Two thoughts:

1. Let's just make the underline below `The rockin' roc repl` be the same color as the words `roc repl`, because the two colors there don't necessarily seem to go together well depending on the terminal.
2. Does this still work on Windows? We've seen problems in the past where Windows terminal is missing features that other terminals have, which has been a problem sometimes.

**Comment by lukewilliamboswell** [comment]
> 2. Does this still work on Windows? We've seen problems in the past where Windows terminal is missing features that other terminals have, which has been a problem sometimes.

This is what is looks like on Windows. Looks good to me. ❤️ 

image

p.s. I'm not sure what is going on with that debug statement in rustyline...

**Review by rtfeldman — APPROVED** [review]

**Comment by rtfeldman** [comment]
Great stuff, thanks for improving this @Anton-4! 😍

**Pull request merged**